streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBRangeIteratorTest.java

```diff
@@ -0,0 +1,419 @@
+                storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key1Bytes,
+            key3Bytes,
+            true
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key2Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key3Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldReturnAllKeysInPartiallyOverlappingRangeInReverseDirection() {
+
+        final RocksIterator rocksIterator = mock(RocksIterator.class);
+        final String to = "e";
+        final Bytes toBytes = Bytes.wrap(to.getBytes());
+        rocksIterator.seekForPrev(toBytes.get());
+        expect(rocksIterator.isValid())
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(false);
+        expect(rocksIterator.key())
+            .andReturn(key4Bytes.get())
+            .andReturn(key3Bytes.get());
+        expect(rocksIterator.value()).andReturn(valueBytes).times(2);
+        rocksIterator.prev();
+        expectLastCall().times(2);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+                storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key3Bytes,
+            toBytes,
+            false
+        );
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key4Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.next().key, is(key3Bytes));
+        assertThat(rocksDBRangeIterator.hasNext(), is(false));
+        verify(rocksIterator);
+    }
+
+    @Test
+    public void shouldReturnTheCurrentKeyOnInvokingPeekNextKeyInForwardDirection() {
+        final RocksIterator rocksIterator = mock(RocksIterator.class);
+        rocksIterator.seek(key1Bytes.get());
+        expect(rocksIterator.isValid())
+            .andReturn(true)
+            .andReturn(true)
+            .andReturn(false);
+        expect(rocksIterator.key())
+            .andReturn(key2Bytes.get())
+            .andReturn(key3Bytes.get());
+        expect(rocksIterator.value()).andReturn(valueBytes).times(2);
+        rocksIterator.next();
+        expectLastCall().times(2);
+        replay(rocksIterator);
+        final RocksDBRangeIterator rocksDBRangeIterator = new RocksDBRangeIterator(
+                storeName,
+            rocksIterator,
+            Collections.emptySet(),
+            key1Bytes,
+            key3Bytes,
+            true
+        );
+
+        assertThat(rocksDBRangeIterator.hasNext(), is(true));
+        assertThat(rocksDBRangeIterator.peekNextKey(), is(key2Bytes));
+        assertThat(rocksDBRangeIterator.next().key, is(key2Bytes));
```

Comment:
I think, it would be good to verify that a second call to  `peekNextKey()` right after the first call to `peekNextKey()` returns the same value, since this is the main difference between `next()` and `peekNextKey()`.